src/main/java/frc/robot/subsystems/ChassisSubsystem.java

```diff
@@ -151,30 +152,41 @@ public void arcadeDrive(double speed, double rotation) {
    * @param scale
    */
   public void drive(RobotContainer robotContainer, DriverOI driverOI, int scale) {
+    String driveTrainType = Preferences.getString(Constants.DRIVE_TRAIN_TYPE, "DEFAULT");
```

Comment:
There is no reason to read the preference every 20ms. Please do not check in this code.

Reply:
There is also no reason not to. Looking up a preference is a cheap operation. There is no good place to place the lookup in the init functions, except to put them into all of them which is a needlessly complex approach.  A better approach would have been to use selectCommand and do the lookup with the command is instantiated each time, but I didn't want to make major changes to the structure of the code. Doing it in a periodic function is not uncommon, you can find quite a few instances of this used in tuning PID commands.

Reply:
I already have working code and it is working in disable_init. There is no need to query in periodic. The drive team needs just to switch it to what they want and be done with it.